hyperspy/io_plugins/ripple.py

```diff
@@ -521,16 +532,11 @@ def file_writer(filename, signal, encoding='latin-1', *args, **kwds):
     # Check if the dtype is supported
     dc = signal.data
     dtype_name = signal.data.dtype.name
+    dimension = len(signal.data.shape)
     if dtype_name not in dtype2keys.keys():
         err = 'The ripple format does not support writting data of %s type' % (
             dtype_name)
         raise IOError(err)
-    # Check if the dimensions are supported
```

Comment:
I think, this part of the code is never used, because it is filtered in the [hyperspy.io](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/io.py#L422)